src/Amadevus.RecordGenerator.Attributes/Features.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace Amadevus.RecordGenerator
+{
+    /// <summary>
+    /// Specifies record features that should be generated.
+    /// </summary>
+    [Flags]
+    public enum Features
+    {
+        /// <summary>
+        /// Zero value.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Record constructor that takes and assigns all read-only properties.
+        /// </summary>
+        Constructor = 0b_1,
+
+        /// <summary>
+        /// With* method per property that creates new record instance with
+        /// provided value assigned to the associated property. Also includes
+        /// generation of Update method.
+        /// </summary>
+        WithPerProperty = 0b_10,
+
+        /// <summary>
+        /// Basic features: <see cref="Constructor"/> and <see cref="WithPerProperty"/>.
+        /// </summary>
+        Basic = Constructor | WithPerProperty,
```

Comment:
`Basic` could mean different things to different people so I wouldn't go there; risky business and adds little value. It pops all kinds of confusing questions like why isn't `Default` the same as `Basic` and which one is more feature-rich? It would require someone to keep looking up the definitions. Also, since `Default` _is_ more feature-rich as defined now, it gives the impression that `Basic` was needed because `Default` could be too beefy and that evokes doubt and distrust. I would just stick with `Default` as the recommended set from the project, which should frankly, give the closest implementation to [how records may get done in C#](https://github.com/dotnet/csharplang/blob/master/proposals/records.md). Again, a project can define its own `Features` constants for their record flavours/schemes/themes.

Reply:
I agree. I'm not highly inclined to keep it, but still making a simple way to have ctor+withs (simpler than OR-ing it) is something that might be useful. `Barebones`? `Minimal`? Or maybe nothing indeed.

Reply:
> `Barebones`? `Minimal`?

Why not let a project define those?

> Or maybe nothing indeed.

Better than getting it wrong because I feel we might be over-thinking the problem. 95% of the time, `Default` should be good enough. The other 5% have a reasonable workaround (BYOF or bring your own `Features` constants). Unless you see those percentages differently?

Reply:
I don't see those percentages. Which convinces me to remove that. :)

Reply:
Removed.